app/src/features/projects/components/ProjectLocationForm.tsx

```diff
@@ -89,7 +96,37 @@ const ProjectLocationForm: React.FC<IProjectLocationFormProps> = (props) => {
     ]);
   }, [values.geometry]);
 
-  const handleSpatialUpload = async (e: any) => {
+  /**
+   * Convert a zipped shapefile to geojson
+   * @param e The file upload event
+   */
+  const handleShapefileUpload = (e: any) => {
+    // Only accept one file
+    const file = e.target.files[0];
+
+    // Back out if not a zipped file
+    if (!file?.type.match(/zip/)) {
+      return;
+    }
+
+    // Create a file reader to extract the binary data
+    const reader = new FileReader();
+    reader.readAsArrayBuffer(file);
+
+    // When the file is loaded run the conversion
+    reader.onload = (event: any) => {
+      // The converter wants a buffer
+      const zip: Buffer = event?.target?.result as Buffer;
+
+      // Run the conversion
+      shp(zip).then((geojson: any) => {
+        const features = (geojson as any).features;
+        setFieldValue('geometry', [...features, ...values.geometry]);
+      });
```

Comment:
Can you change this to use async/await?

There could be some cases where regular .then style promises are useful, but in general prefer async/await style.